RELEASE.md

```diff
@@ -0,0 +1,41 @@
+# RELEASE PROCESS
+
+This document outlines the process for creating releases for
+`gke-networking-api`, including versioning conventions and a changelog of notable
+changes in each release.
+
+## Module Versioning
+
+When merging changes, please request a tag update from an owner in OWNERS
+for the merged changes when needed.
+
+This project uses Go module versioning
+(https://go.dev/doc/modules/version-numbers), which uses semantic versioning
+(vMAJOR.MINOR.PATCH) to indicate the nature of changes.
+
+However, as the project has the v0 major version, so it makes no stability or
+backward compatibility guarantees.
+
+* MINOR: Adding functionality.
+* PATCH: Bug fixes.
+
+## Releases
+
+**Currently, release tags are created manually from the CLI using the following commands:**
+
+```sh
+git tag <tag>
+git push origin <tag>
+```
+
+### v0.1.1 (Jul 2, 2024)
```

Comment:
You should create a CHANGELOG for this information.

This looks like a reasonable description of what should do: https://keepachangelog.com/en/1.1.0/

probably link that here and add to the procedure.

Reply:
Done. I also populated it with the major updates done to the repo, so I think it's in a good shape now. Could you review it please?

Reply:
Oh, sorry, wait a bit please, the changes were not pushed for some reason... I'm looking into it.

Reply:
Done.